Enhanced multiselect dropdowns (SHOOP-2324)

tulimaki opened the pull request:
Admin: Add view and widget to support Select2 dropdowns
Admin: Enable multiselect dropdown with Select2
Admin: Remove support for selectpicker
Campaigns: Use Select2 for ManyToManyField's

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1854/
Test FAILed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1855/
Test FAILed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1856/
Test FAILed.

tulimaki commented:
retest this please

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1858/
Test PASSed.

suutari-ai commented:
Why support for select pickers was removed? (Commit message doesn't say anything.)

suutari-ai commented on shoop/campaigns/forms.py:
```diff
@@ -77,13 +81,33 @@ def __init__(self, *args, **kwargs):
+            if not isinstance(initial_value, str):
+                initial_value = [o.pk for o in initial_value]
+            else:
+                initial_value = [initial_value]
```
Got `TypeError: 'decimal.Decimal' object is not iterable` from the `initial_value = [o.pk for o in initial_value]` line of this.

suutari-ai replied:
How about this one instead?
```python
    def _get_initial_value(self, obj):
        initial_value = self.get_initial_value(obj.identifier)
        if initial_value is None:
            return None
        elif isinstance(initial_value, QuerySet):
            return [x.pk for x in initial_value]
        else:
            return [initial_value]
```

suutari-ai commented:
* IMHO Products should be selectable by SKU too (and probably SKU should be shown in the selected item and in the dropdown too)
* The select2 fields should be styled (especially since all select fields are now using the select2):
   * Focus color should be same as for other input fields.
   * Padding should be fixed to be same above and below the items.
   * Select 2 fields should have same weight as other fields (when they fit to single line)

tulimaki commented:
* Added better styles for select2 selects
* Added option to search with SKU.
* Changed object text to force_text(obj) instead of objecte name. I think it's up to Product __str__ to display sku. And I think it would add unneccesary complexety to decide display based on the object type.
* Added fix to _get_initial_value in campaign forms

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1863/
Test PASSed.
